### PR TITLE
Add epoch time as a column in throughput calculations

### PR DESCRIPTION
**Pull request by patrickherring-TRI**
This PR adds a column "epoch_time" that is the timestamp in seconds of the cycle metric that is being calculated. This should make it possible to use epoch time in the threshold calculations

### CODE REVIEW

**Review by moorepatrick — COMMENTED** [review]
I'd like to let someone else who uses this code give the approval, but is the x_axis calculations commit related?

**Comment by patrickherring-TRI** [comment]
@moorepatrick Yes, the updated x_axis calculation will use the time basis that is being added to the features.

**Review by pasinger — APPROVED** [review]
LGTM!

**Review by pasinger — APPROVED** [review]
LGTM

File: beep/dataset.py [hunk 1]
```diff
@@ -400,39 +408,45 @@ def get_threshold_targets(dataset_diagnostic_properties,
         elif run_target_df['fractional_metric'].min() > threshold and extrapolate_threshold:
             fill_value = "extrapolate"
             bounds_error = False
-            x_throughput_linspace = np.linspace(x_throughput_axis.min(), 2 * x_throughput_axis.max(), num=1000)
-            x_cycle_linspace = np.linspace(x_cycle_axis.min(), 2 * x_cycle_axis.max(), num=1000)
+            x_linspaces = []
+            for axis in x_axes:
+                x_linspaces.append(np.linspace(axis.min(), 2 * axis.max(), num=1000))
```
Comment (pasinger) [inline beep/dataset.py:413]: Why does the linspace upper bound go to 2*axis.max() instead of just axis.max()?

Comment (pasinger) [reply beep/dataset.py:413]: Ah I see now that this is for cases where the fractional_metric has not yet crossed the threshold. I see why there should be a factor greater than 1, but a value of 2 feels arbitrary.
If a test is very early in its lifetime and extrapolation is on, then 2*axis.max() may still not provide a value that will go below the threshold.
Could it be made more specific such as by making the upper bound the predicted linear extrapolation:
x_thresh_extrap = (threshold - y1)*(x2-x1)/(y2-y1)+x1, where
 y1 = y_interpolation_axis.iloc[-2]
y2 = y_interpolation_axis.iloc[-1]
x1 = axis[-2]
x2 = axis[-1]
We could make the extrapolation go a little beyond what's exactly necessary by subtracting another 0.1 from threshold, for example.

File: beep/dataset.py [hunk 2]
```diff
@@ -400,39 +408,45 @@ def get_threshold_targets(dataset_diagnostic_properties,
...
-            x_cycle_linspace = np.linspace(x_cycle_axis.min(), x_cycle_axis.max(), num=1000)
+            x_linspaces = []
+            for axis in x_axes:
+                x_linspaces.append(np.linspace(axis.min(), axis.max(), num=1000))
 
-        f_throughput = interpolate.interp1d(x_throughput_axis, y_interpolation_axis, kind='linear',
-                                            bounds_error=bounds_error, fill_value=fill_value)
-        f_cycle = interpolate.interp1d(x_cycle_axis, y_interpolation_axis, kind='linear', bounds_error=bounds_error,
-                                       fill_value=fill_value)
+        f_axis = []
```
Comment (pasinger) [inline beep/dataset.py:421]: Can we move the this initialization of f_axis to below that of x_linspaces, then combine the below code (second for loop over x_axes) to with the first for loop over x_axes?

**Review by pasinger — COMMENTED** [review]
This looks like a nice update to enable the retrieval of more types of measured values at the threshold value. I made a few suggestions, but it looks to me like it will work just as well as it was previously without making the suggested changes.

**Pull request merged**